File: src/upros_lidar_filter/src/move_lidar.cpp
#include <ros/ros.h>
#include <sensor_msgs/LaserScan.h>
#include <geometry_msgs/Twist.h>

ros::Publisher cmdVelPub;
float firstRange, lastRange;

void scanCallback(const sensor_msgs::LaserScan::ConstPtr &scan)
{
    int num_points = (scan->angle_max - scan->angle_min) / scan->angle_increment;
    firstRange = scan->ranges[0];
    lastRange = scan->ranges[num_points - 1];
    float meanRange = (firstRange + lastRange) / 2.0;
    ROS_INFO_STREAM("Mean range: " << meanRange);

    geometry_msgs::Twist cmdVelMsg;
    if (meanRange >= 8.0)
    {
        cmdVelMsg.linear.x = 0.0;
    }
    else
    {
        cmdVelMsg.linear.x = 0.3;
    }
    cmdVelPub.publish(cmdVelMsg);
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "laser_scan_listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("/scan", 1000, scanCallback);
    cmdVelPub = n.advertise<geometry_msgs::Twist>("/cmd_vel", 10);
    ros::spin();
    return 0;
}

File: src/upros_lidar_filter/scripts/move_lidar.py
#!/usr/bin/env python3

import rospy
from sensor_msgs.msg import LaserScan
from geometry_msgs.msg import Twist

def scanCallback(scan):
    num_points = int((scan.angle_max - scan.angle_min) / scan.angle_increment)
    first_range = scan.ranges[0]
    last_range = scan.ranges[num_points - 1]
    mean_range = (first_range + last_range) / 2.0
    rospy.loginfo("Mean range: %f", mean_range)

    cmd_vel_msg = Twist()
    if mean_range >= 8.0:
        cmd_vel_msg.linear.x = 0.0
    else:
        cmd_vel_msg.linear.x = 0.3
    cmd_vel_pub.publish(cmd_vel_msg)

rospy.init_node('laser_scan_listener', anonymous=True)
cmd_vel_pub = rospy.Publisher('/cmd_vel', Twist, queue_size=10)
sub = rospy.Subscriber('/scan', LaserScan, scanCallback)
rospy.spin()